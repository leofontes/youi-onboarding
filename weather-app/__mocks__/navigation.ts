export const navigationProp = {
  navigation: {
    addListener: jest.fn(),
    isFocused: jest.fn(),
    navigate: jest.fn(),
    getParam: jest.fn(),
  },
};
