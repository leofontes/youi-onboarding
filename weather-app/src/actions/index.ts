import * as tmdbActions from './tmdbActions';
import * as youtubeActions from './youtubeActions';

export {
  tmdbActions as tmdb,
  youtubeActions as youtube,
};
