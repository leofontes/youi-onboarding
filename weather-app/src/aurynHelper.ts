import { DeviceInfo, Ref } from '@youi/react-native-youi';
import { NativeModules, findNodeHandle } from 'react-native';
import { once } from 'lodash';

const systemName = DeviceInfo.getSystemName();
const { Cloud, RefUtils } = NativeModules;

interface AurynHelper {
  hasHardwareBackButton: boolean;
  isRoku: boolean;
  updateCloudScene: (component: React.RefObject<any>) => void;
  togglePointerEvents: (ref: Ref, enabled: boolean) => boolean;
  toggleParentPointerEvents: (ref: Ref, enabled: boolean) => boolean;
}

export const AurynHelper: AurynHelper = {
  hasHardwareBackButton: !['iOS'].includes(systemName),
  isRoku: Cloud.isCloudServer,
  updateCloudScene: (component) => {
    if (!Cloud.isCloudServer || !component.current) return;
    setTimeout(() => {
      Cloud.exportSubtree(findNodeHandle(component.current), false);
      Cloud.sendFocusMap();
    }, 300);
  },
  togglePointerEvents: once((ref: Ref, enabled: boolean) => {
    RefUtils.setPointerEvents(findNodeHandle(ref), enabled);
    return true;
  }),
  toggleParentPointerEvents: once((ref: Ref, enabled: boolean) => {
    RefUtils.setParentCompositionPointerEvents(findNodeHandle(ref), enabled);
    return true;
  }),
};
