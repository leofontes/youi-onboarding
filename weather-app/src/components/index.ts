export { BackButton } from './backButton';
export { DiscoverContainer } from './discoverContainer';
export { List } from './list';
export { ListItem } from './listitem';
export { Timeline } from './timeline';
export { ToggleButton } from './toggleButton';
export { ToggleGroup } from './toggleGroup';
export { NavigationBar } from './navigationBar';
export { TvContainer } from './tvContainer';
export { Error } from './error';
export { withOrientation } from './withOrientation';
export { LiveContainer } from './liveContainer';
export { LiveListItem } from './liveListitem';
export { BackgroundImage } from './backgroundImage';
