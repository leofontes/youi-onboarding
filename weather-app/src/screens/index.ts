export { Splash } from './splash';
export { Lander } from './lander';
export { Pdp } from './pdp';
export { Search } from './search';
export { Profile } from './profile';
export { VideoScreen as Video } from './video';
export { AdOverlay } from './../components/adManager/list-item-ad';
