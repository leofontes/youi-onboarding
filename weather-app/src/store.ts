import promise from 'redux-promise-middleware';
import createDebounce from 'redux-debounced';
import thunk from 'redux-thunk';
import { rootReducer, StoreState } from './reducers';

import { configureStore, ConfigureStoreOptions, AnyAction } from '@reduxjs/toolkit';

export default configureStore({
  reducer: rootReducer,
  middleware: [createDebounce(), thunk, promise]
} as ConfigureStoreOptions<StoreState, AnyAction>);
