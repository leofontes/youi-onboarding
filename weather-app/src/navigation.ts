import * as Screens from './screens';
import { createStackNavigator, createSwitchNavigator, StackNavigatorConfig } from 'react-navigation';
import { AurynHelper } from './aurynHelper';

const stackOptions: StackNavigatorConfig = {
  headerMode: 'none',
  cardStyle: {
    backgroundColor: 'transparent',
    opacity: 1,
  },
  transitionConfig: () => ({
    transitionSpec: {
      duration: 0,
    },
    containerStyle: {
      backgroundColor: 'transparent',
    },
    screenInterpolator: () => {},
  }),
};

const createNavigator = AurynHelper.isRoku ? createSwitchNavigator : createStackNavigator;
const AppStack = createNavigator(
  {
    Lander: { screen: Screens.Lander },
    PDP: { screen: Screens.Pdp },
    Search: { screen: Screens.Search },
    Profile: { screen: Screens.Profile },
    Video: { screen: Screens.Video },
    AdOverlay: { screen: Screens.AdOverlay },
  },
  {
    ...stackOptions,
    initialRouteName: 'Lander',
  },
);

const SplashStack = createStackNavigator(
  { Splash: { screen: Screens.Splash } },
  {
    ...stackOptions,
    initialRouteName: 'Splash',
  },
);

const rootNavigationStack = createSwitchNavigator({
  Splash: SplashStack,
  App: AppStack,
});

export default rootNavigationStack;
