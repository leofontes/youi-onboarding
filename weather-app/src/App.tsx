import React from 'react';
import store from './store';
import Stack from './navigation';
import { FormFactor } from '@youi/react-native-youi';
import { Provider } from 'react-redux';
import { BackgroundImage } from './components';

export default class YiReactApp extends React.Component {
  render = () =>
    <Provider store={store}>
      <BackgroundImage source={!FormFactor.isHandset ? 'res://drawable/default/CES-auryn-10ft-bg.png' : 'res://drawable/default/Gradient-Bottom-2.png'}>
        <Stack />
      </BackgroundImage>
    </Provider>
}
