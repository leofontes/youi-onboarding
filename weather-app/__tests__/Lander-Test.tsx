
import React from 'react';
import { Lander } from '../src/screens';
import { Provider } from 'react-redux';
import { navigationProp } from '../__mocks__/navigation';
import store from '../src/store';
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<Provider store={store}>
    <Lander {...navigationProp} />
  </Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
