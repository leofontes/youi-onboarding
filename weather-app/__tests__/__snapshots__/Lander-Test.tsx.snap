// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Composition
  onTimelinesLoaded={[Function]}
  source="Auryn_Lander"
>
  <ListRef
    DEPRECATED_sendUpdatedChildFrames={false}
    alwaysBounceHorizontal={true}
    alwaysBounceVertical={false}
    horizontal={true}
    initialToggleIndex={0}
    name="Nav-List"
    onContentSizeChange={null}
    onLayout={[MockFunction]}
    onMomentumScrollBegin={[MockFunction]}
    onMomentumScrollEnd={[MockFunction]}
    onPressItem={[Function]}
    onResponderGrant={[MockFunction]}
    onResponderReject={[MockFunction]}
    onResponderRelease={[MockFunction]}
    onResponderTerminationRequest={[MockFunction]}
    onScroll={[MockFunction]}
    onScrollBeginDrag={[MockFunction]}
    onScrollEndDrag={[MockFunction]}
    onScrollShouldSetResponder={[MockFunction]}
    onStartShouldSetResponder={[MockFunction]}
    onStartShouldSetResponderCapture={[MockFunction]}
    onTouchCancel={[MockFunction]}
    onTouchEnd={[MockFunction]}
    onTouchMove={[MockFunction]}
    onTouchStart={[MockFunction]}
    scrollBarThumbImage={null}
    scrollEnabled={false}
    sendMomentumEvents={false}
    snapToEnd={true}
    snapToStart={true}
    style={
      Array [
        Object {
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
          "overflow": "scroll",
        },
        undefined,
      ]
    }
  >
    <ScrollContentData
      collapsable={false}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "minWidth": "100%",
          }
        }
      >
        <Composition
          onTimelinesLoaded={[Function]}
          source="Auryn_Container-Btn-Nav-List"
        >
          <YiPushButtonView
            name="Btn-Nav-List"
            onCompositionDidLoad={[Function]}
            onFocus={[Function]}
            onPress={[Function]}
            onTimelinesLoaded={[Function]}
          >
            <TimelineRef
              autoplay={true}
              direction="forward"
              loop={false}
              name="Toggle-On"
              onAborted={[Function]}
              onCompleted={[Function]}
              onCompletedForward={[Function]}
              onCompletedReverse={[Function]}
              onCompositionDidLoad={[Function]}
              onPaused={[Function]}
              onPausedForward={[Function]}
              onPausedReverse={[Function]}
              onPlay={[Function]}
              onPlayForward={[Function]}
              onPlayReverse={[Function]}
              onStarted={[Function]}
              onStartedForward={[Function]}
              onStartedReverse={[Function]}
            />
            <YiTextSceneNode
              name="title"
              style={
                Object {
                  "color": "#F1F1F1",
                }
              }
              text="Discover"
            />
            <YiImageView
              name="Nav-Icon"
              onTimelinesLoaded={[Function]}
              source={
                Object {
                  "uri": "res://drawable/default/Default-Nav-Icon.png",
                }
              }
            />
            <YiImageView
              name="Nav-Icon-Toggled"
              onTimelinesLoaded={[Function]}
              source={
                Object {
                  "uri": "res://drawable/default/Default-Toggled-Nav-Icon.png",
                }
              }
              style={
                Object {
                  "resizeMode": "contain",
                }
              }
            />
            <YiTextSceneNode
              name="title-toggled"
              text="Discover"
            />
          </YiPushButtonView>
        </Composition>
        <Composition
          onTimelinesLoaded={[Function]}
          source="Auryn_Container-Btn-Nav-List"
        >
          <YiPushButtonView
            name="Btn-Nav-List"
            onCompositionDidLoad={[Function]}
            onFocus={[Function]}
            onPress={[Function]}
            onTimelinesLoaded={[Function]}
          >
            <TimelineRef
              autoplay={false}
              direction="reverse"
              loop={false}
              name="Toggle-On"
              onAborted={[Function]}
              onCompleted={[Function]}
              onCompletedForward={[Function]}
              onCompletedReverse={[Function]}
              onCompositionDidLoad={[Function]}
              onPaused={[Function]}
              onPausedForward={[Function]}
              onPausedReverse={[Function]}
              onPlay={[Function]}
              onPlayForward={[Function]}
              onPlayReverse={[Function]}
              onStarted={[Function]}
              onStartedForward={[Function]}
              onStartedReverse={[Function]}
            />
            <YiTextSceneNode
              name="title"
              style={
                Object {
                  "color": "#F1F1F1",
                }
              }
              text="Movies"
            />
            <YiImageView
              name="Nav-Icon"
              onTimelinesLoaded={[Function]}
              source={
                Object {
                  "uri": "res://drawable/default/Movie-Nav-Icon.png",
                }
              }
            />
            <YiImageView
              name="Nav-Icon-Toggled"
              onTimelinesLoaded={[Function]}
              source={
                Object {
                  "uri": "res://drawable/default/Movie-Toggled-Nav-Icon.png",
                }
              }
              style={
                Object {
                  "resizeMode": "contain",
                }
              }
            />
            <YiTextSceneNode
              name="title-toggled"
              text="Movies"
            />
          </YiPushButtonView>
        </Composition>
        <Composition
          onTimelinesLoaded={[Function]}
          source="Auryn_Container-Btn-Nav-List"
        >
          <YiPushButtonView
            name="Btn-Nav-List"
            onCompositionDidLoad={[Function]}
            onFocus={[Function]}
            onPress={[Function]}
            onTimelinesLoaded={[Function]}
          >
            <TimelineRef
              autoplay={false}
              direction="reverse"
              loop={false}
              name="Toggle-On"
              onAborted={[Function]}
              onCompleted={[Function]}
              onCompletedForward={[Function]}
              onCompletedReverse={[Function]}
              onCompositionDidLoad={[Function]}
              onPaused={[Function]}
              onPausedForward={[Function]}
              onPausedReverse={[Function]}
              onPlay={[Function]}
              onPlayForward={[Function]}
              onPlayReverse={[Function]}
              onStarted={[Function]}
              onStartedForward={[Function]}
              onStartedReverse={[Function]}
            />
            <YiTextSceneNode
              name="title"
              style={
                Object {
                  "color": "#F1F1F1",
                }
              }
              text="Shows"
            />
            <YiImageView
              name="Nav-Icon"
              onTimelinesLoaded={[Function]}
              source={
                Object {
                  "uri": "res://drawable/default/Series-Nav-Icon.png",
                }
              }
            />
            <YiImageView
              name="Nav-Icon-Toggled"
              onTimelinesLoaded={[Function]}
              source={
                Object {
                  "uri": "res://drawable/default/Series-Toggled-Nav-Icon.png",
                }
              }
              style={
                Object {
                  "resizeMode": "contain",
                }
              }
            />
            <YiTextSceneNode
              name="title-toggled"
              text="Shows"
            />
          </YiPushButtonView>
        </Composition>
        <Composition
          onTimelinesLoaded={[Function]}
          source="Auryn_Container-Btn-Nav-List"
        >
          <YiPushButtonView
            name="Btn-Nav-List"
            onCompositionDidLoad={[Function]}
            onFocus={[Function]}
            onPress={[Function]}
            onTimelinesLoaded={[Function]}
          >
            <TimelineRef
              autoplay={false}
              direction="reverse"
              loop={false}
              name="Toggle-On"
              onAborted={[Function]}
              onCompleted={[Function]}
              onCompletedForward={[Function]}
              onCompletedReverse={[Function]}
              onCompositionDidLoad={[Function]}
              onPaused={[Function]}
              onPausedForward={[Function]}
              onPausedReverse={[Function]}
              onPlay={[Function]}
              onPlayForward={[Function]}
              onPlayReverse={[Function]}
              onStarted={[Function]}
              onStartedForward={[Function]}
              onStartedReverse={[Function]}
            />
            <YiTextSceneNode
              name="title"
              style={
                Object {
                  "color": "#F1F1F1",
                }
              }
              text="Live"
            />
            <YiImageView
              name="Nav-Icon"
              onTimelinesLoaded={[Function]}
              source={
                Object {
                  "uri": "res://drawable/default/Live-Nav-Icon.png",
                }
              }
            />
            <YiImageView
              name="Nav-Icon-Toggled"
              onTimelinesLoaded={[Function]}
              source={
                Object {
                  "uri": "res://drawable/default/Live-Toggled-Nav-Icon.png",
                }
              }
              style={
                Object {
                  "resizeMode": "contain",
                }
              }
            />
            <YiTextSceneNode
              name="title-toggled"
              text="Live"
            />
          </YiPushButtonView>
        </Composition>
      </View>
    </ScrollContentData>
  </ListRef>
  <YiPushButtonView
    name="Btn-Nav-Search"
    onCompositionDidLoad={[Function]}
    onPress={[Function]}
    onTimelinesLoaded={[Function]}
  />
  <YiPushButtonView
    name="Btn-Nav-Profile"
    onCompositionDidLoad={[Function]}
    onPress={[Function]}
    onTimelinesLoaded={[Function]}
  />
  <TimelineRef
    loop={false}
    name="LanderIn"
    onAborted={[Function]}
    onCompleted={[Function]}
    onCompletedForward={[Function]}
    onCompletedReverse={[Function]}
    onCompositionDidLoad={[Function]}
    onPaused={[Function]}
    onPausedForward={[Function]}
    onPausedReverse={[Function]}
    onPlay={[Function]}
    onPlayForward={[Function]}
    onPlayReverse={[Function]}
    onStarted={[Function]}
    onStartedForward={[Function]}
    onStartedReverse={[Function]}
  />
  <TimelineRef
    loop={false}
    name="LanderOut"
    onAborted={[Function]}
    onCompleted={[Function]}
    onCompletedForward={[Function]}
    onCompletedReverse={[Function]}
    onCompositionDidLoad={[Function]}
    onPaused={[Function]}
    onPausedForward={[Function]}
    onPausedReverse={[Function]}
    onPlay={[Function]}
    onPlayForward={[Function]}
    onPlayReverse={[Function]}
    onStarted={[Function]}
    onStartedForward={[Function]}
    onStartedReverse={[Function]}
  />
  <ListRef
    DEPRECATED_sendUpdatedChildFrames={false}
    alwaysBounceHorizontal={false}
    alwaysBounceVertical={true}
    data={
      Array [
        <_class
          source="Auryn_Container-Lander-List"
        >
          <List
            data={Array []}
            name="Lander-List"
            onFocusItem={[Function]}
            onPressItem={[Function]}
            type={0}
          />
        </_class>,
        <_class
          source="Auryn_Container-Lander-List"
          style={
            Object {
              "marginTop": 0,
            }
          }
        >
          <List
            data={Array []}
            name="Lander-List"
            onFocusItem={[Function]}
            onPressItem={[Function]}
            snapToAlignment="center"
            type={1}
          />
        </_class>,
        <_class
          source="Auryn_Container-Lander-List"
        >
          <List
            data={Array []}
            name="Lander-List"
            onFocusItem={[Function]}
            onPressItem={[Function]}
            type={2}
          />
        </_class>,
        <_class
          source="Auryn_Container-Live"
          style={
            Object {
              "marginTop": 0,
            }
          }
        >
          <List
            data={Array []}
            name="LiveList"
            onFocusItem={[Function]}
            onPressItem={[Function]}
            type={6}
          />
        </_class>,
      ]
    }
    disableVirtualization={false}
    focusable={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    name="Stack"
    numColumns={1}
    onContentSizeChange={null}
    onEndReachedThreshold={2}
    onLayout={[MockFunction]}
    onMomentumScrollBegin={[MockFunction]}
    onMomentumScrollEnd={[MockFunction]}
    onResponderGrant={[MockFunction]}
    onResponderReject={[MockFunction]}
    onResponderRelease={[MockFunction]}
    onResponderTerminationRequest={[MockFunction]}
    onScroll={[MockFunction]}
    onScrollBeginDrag={[MockFunction]}
    onScrollEndDrag={[MockFunction]}
    onScrollShouldSetResponder={[MockFunction]}
    onStartShouldSetResponder={[MockFunction]}
    onStartShouldSetResponderCapture={[MockFunction]}
    onTouchCancel={[MockFunction]}
    onTouchEnd={[MockFunction]}
    onTouchMove={[MockFunction]}
    onTouchStart={[MockFunction]}
    onViewableItemsChanged={[Function]}
    renderItem={[Function]}
    renderScrollComponent={[Function]}
    scrollBarThumbImage={null}
    scrollEnabled={false}
    scrollEventThrottle={50}
    sendMomentumEvents={true}
    snapToEnd={true}
    snapToStart={true}
    stickyHeaderIndices={Array []}
    style={
      Array [
        Object {
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 1,
          "overflow": "scroll",
        },
        undefined,
      ]
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={
      Array [
        Object {
          "onViewableItemsChanged": [Function],
          "viewabilityConfig": undefined,
        },
      ]
    }
    windowSize={21}
  >
    <ScrollContentData
      collapsable={false}
      onLayout={[MockFunction]}
      style={
        Array [
          false,
          undefined,
        ]
      }
    >
      <View
        onLayout={[Function]}
        style={null}
      >
        <Composition
          onTimelinesLoaded={[Function]}
          source="Auryn_Container-Lander-List"
        >
          <ListRef
            DEPRECATED_sendUpdatedChildFrames={false}
            alwaysBounceHorizontal={true}
            alwaysBounceVertical={false}
            data={Array []}
            disableVirtualization={false}
            getItem={[Function]}
            getItemCount={[Function]}
            getItemLayout={[Function]}
            horizontal={true}
            initialNumToRender={2}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            name="Lander-List"
            numColumns={1}
            onContentSizeChange={null}
            onEndReachedThreshold={2}
            onLayout={[MockFunction]}
            onMomentumScrollBegin={[MockFunction]}
            onMomentumScrollEnd={[MockFunction]}
            onResponderGrant={[MockFunction]}
            onResponderReject={[MockFunction]}
            onResponderRelease={[MockFunction]}
            onResponderTerminationRequest={[MockFunction]}
            onScroll={[MockFunction]}
            onScrollBeginDrag={[MockFunction]}
            onScrollEndDrag={[MockFunction]}
            onScrollShouldSetResponder={[MockFunction]}
            onStartShouldSetResponder={[MockFunction]}
            onStartShouldSetResponderCapture={[MockFunction]}
            onTouchCancel={[MockFunction]}
            onTouchEnd={[MockFunction]}
            onTouchMove={[MockFunction]}
            onTouchStart={[MockFunction]}
            renderItem={[Function]}
            renderScrollComponent={[Function]}
            scrollBarThumbImage={null}
            scrollEnabled={true}
            scrollEventThrottle={50}
            sendMomentumEvents={true}
            snapToEnd={true}
            snapToStart={true}
            stickyHeaderIndices={Array []}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "overflow": "scroll",
                },
                undefined,
              ]
            }
            updateCellsBatchingPeriod={50}
            viewabilityConfigCallbackPairs={Array []}
            windowSize={21}
          >
            <ScrollContentData
              collapsable={false}
              onLayout={[MockFunction]}
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            />
          </ListRef>
          <ViewRef
            name="Lander-List"
            onCompositionDidLoad={[Function]}
            onTimelinesLoaded={[Function]}
          />
        </Composition>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Composition
          onTimelinesLoaded={[Function]}
          source="Auryn_Container-Lander-List"
          style={
            Object {
              "marginTop": 0,
            }
          }
        >
          <ListRef
            DEPRECATED_sendUpdatedChildFrames={false}
            ListHeaderComponent={[Function]}
            alwaysBounceHorizontal={true}
            alwaysBounceVertical={false}
            data={Array []}
            disableVirtualization={false}
            getItem={[Function]}
            getItemCount={[Function]}
            getItemLayout={[Function]}
            horizontal={true}
            initialNumToRender={2}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            name="Lander-List"
            numColumns={1}
            onContentSizeChange={null}
            onEndReachedThreshold={2}
            onLayout={[MockFunction]}
            onMomentumScrollBegin={[MockFunction]}
            onMomentumScrollEnd={[MockFunction]}
            onResponderGrant={[MockFunction]}
            onResponderReject={[MockFunction]}
            onResponderRelease={[MockFunction]}
            onResponderTerminationRequest={[MockFunction]}
            onScroll={[MockFunction]}
            onScrollBeginDrag={[MockFunction]}
            onScrollEndDrag={[MockFunction]}
            onScrollShouldSetResponder={[MockFunction]}
            onStartShouldSetResponder={[MockFunction]}
            onStartShouldSetResponderCapture={[MockFunction]}
            onTouchCancel={[MockFunction]}
            onTouchEnd={[MockFunction]}
            onTouchMove={[MockFunction]}
            onTouchStart={[MockFunction]}
            renderItem={[Function]}
            renderScrollComponent={[Function]}
            scrollBarThumbImage={null}
            scrollEnabled={true}
            scrollEventThrottle={50}
            sendMomentumEvents={true}
            snapToAlignment="center"
            snapToEnd={true}
            snapToStart={true}
            stickyHeaderIndices={Array []}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "overflow": "scroll",
                },
                undefined,
              ]
            }
            updateCellsBatchingPeriod={50}
            viewabilityConfigCallbackPairs={Array []}
            windowSize={21}
          >
            <ScrollContentData
              collapsable={false}
              onLayout={[MockFunction]}
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            >
              <View
                onLayout={[Function]}
              />
            </ScrollContentData>
          </ListRef>
          <ViewRef
            name="Lander-List"
            onCompositionDidLoad={[Function]}
            onTimelinesLoaded={[Function]}
          />
        </Composition>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Composition
          onTimelinesLoaded={[Function]}
          source="Auryn_Container-Lander-List"
        >
          <ListRef
            DEPRECATED_sendUpdatedChildFrames={false}
            alwaysBounceHorizontal={true}
            alwaysBounceVertical={false}
            data={Array []}
            disableVirtualization={false}
            getItem={[Function]}
            getItemCount={[Function]}
            getItemLayout={[Function]}
            horizontal={true}
            initialNumToRender={2}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            name="Lander-List"
            numColumns={1}
            onContentSizeChange={null}
            onEndReachedThreshold={2}
            onLayout={[MockFunction]}
            onMomentumScrollBegin={[MockFunction]}
            onMomentumScrollEnd={[MockFunction]}
            onResponderGrant={[MockFunction]}
            onResponderReject={[MockFunction]}
            onResponderRelease={[MockFunction]}
            onResponderTerminationRequest={[MockFunction]}
            onScroll={[MockFunction]}
            onScrollBeginDrag={[MockFunction]}
            onScrollEndDrag={[MockFunction]}
            onScrollShouldSetResponder={[MockFunction]}
            onStartShouldSetResponder={[MockFunction]}
            onStartShouldSetResponderCapture={[MockFunction]}
            onTouchCancel={[MockFunction]}
            onTouchEnd={[MockFunction]}
            onTouchMove={[MockFunction]}
            onTouchStart={[MockFunction]}
            renderItem={[Function]}
            renderScrollComponent={[Function]}
            scrollBarThumbImage={null}
            scrollEnabled={true}
            scrollEventThrottle={50}
            sendMomentumEvents={true}
            snapToEnd={true}
            snapToStart={true}
            stickyHeaderIndices={Array []}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "overflow": "scroll",
                },
                undefined,
              ]
            }
            updateCellsBatchingPeriod={50}
            viewabilityConfigCallbackPairs={Array []}
            windowSize={21}
          >
            <ScrollContentData
              collapsable={false}
              onLayout={[MockFunction]}
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            />
          </ListRef>
          <ViewRef
            name="Lander-List"
            onCompositionDidLoad={[Function]}
            onTimelinesLoaded={[Function]}
          />
        </Composition>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Composition
          onTimelinesLoaded={[Function]}
          source="Auryn_Container-Live"
          style={
            Object {
              "marginTop": 0,
            }
          }
        >
          <ListRef
            DEPRECATED_sendUpdatedChildFrames={false}
            alwaysBounceHorizontal={false}
            alwaysBounceVertical={true}
            data={Array []}
            disableVirtualization={false}
            getItem={[Function]}
            getItemCount={[Function]}
            getItemLayout={[Function]}
            horizontal={false}
            initialNumToRender={2}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            name="LiveList"
            numColumns={1}
            onContentSizeChange={null}
            onEndReachedThreshold={2}
            onLayout={[MockFunction]}
            onMomentumScrollBegin={[MockFunction]}
            onMomentumScrollEnd={[MockFunction]}
            onResponderGrant={[MockFunction]}
            onResponderReject={[MockFunction]}
            onResponderRelease={[MockFunction]}
            onResponderTerminationRequest={[MockFunction]}
            onScroll={[MockFunction]}
            onScrollBeginDrag={[MockFunction]}
            onScrollEndDrag={[MockFunction]}
            onScrollShouldSetResponder={[MockFunction]}
            onStartShouldSetResponder={[MockFunction]}
            onStartShouldSetResponderCapture={[MockFunction]}
            onTouchCancel={[MockFunction]}
            onTouchEnd={[MockFunction]}
            onTouchMove={[MockFunction]}
            onTouchStart={[MockFunction]}
            renderItem={[Function]}
            renderScrollComponent={[Function]}
            scrollBarThumbImage={null}
            scrollEnabled={true}
            scrollEventThrottle={50}
            sendMomentumEvents={true}
            snapToEnd={true}
            snapToStart={true}
            stickyHeaderIndices={Array []}
            style={
              Array [
                Object {
                  "flexDirection": "column",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "overflow": "scroll",
                },
                undefined,
              ]
            }
            updateCellsBatchingPeriod={50}
            viewabilityConfigCallbackPairs={Array []}
            windowSize={21}
          >
            <ScrollContentData
              collapsable={false}
              onLayout={[MockFunction]}
              style={
                Array [
                  false,
                  undefined,
                ]
              }
            />
          </ListRef>
          <ViewRef
            name="LiveList"
            onCompositionDidLoad={[Function]}
            onTimelinesLoaded={[Function]}
          />
        </Composition>
      </View>
    </ScrollContentData>
  </ListRef>
  <ViewRef
    name="Stack"
    onCompositionDidLoad={[Function]}
    onTimelinesLoaded={[Function]}
  />
  <ViewRef
    name="Nav"
    onCompositionDidLoad={[Function]}
    onTimelinesLoaded={[Function]}
  >
    <TimelineRef
      loop={false}
      name="In"
      onAborted={[Function]}
      onCompleted={[Function]}
      onCompletedForward={[Function]}
      onCompletedReverse={[Function]}
      onCompositionDidLoad={[Function]}
      onPaused={[Function]}
      onPausedForward={[Function]}
      onPausedReverse={[Function]}
      onPlay={[Function]}
      onPlayForward={[Function]}
      onPlayReverse={[Function]}
      onStarted={[Function]}
      onStartedForward={[Function]}
      onStartedReverse={[Function]}
    />
    <TimelineRef
      loop={false}
      name="Out"
      onAborted={[Function]}
      onCompleted={[Function]}
      onCompletedForward={[Function]}
      onCompletedReverse={[Function]}
      onCompositionDidLoad={[Function]}
      onPaused={[Function]}
      onPausedForward={[Function]}
      onPausedReverse={[Function]}
      onPlay={[Function]}
      onPlayForward={[Function]}
      onPlayReverse={[Function]}
      onStarted={[Function]}
      onStartedForward={[Function]}
      onStartedReverse={[Function]}
    />
  </ViewRef>
</Composition>
`;
